aggregator/src/app/HealthService.ts

```diff
@@ -0,0 +1,192 @@
+import { axiod } from "../../deps.ts";
+import BundleTable from "./BundleTable.ts";
+import AppEvent from "./AppEvent.ts";
+import assert from "../helpers/assert.ts";
+import * as env from "../env.ts";
+
+abstract class HealthIndicator {
+  constructor(
+    public emit: (evt: AppEvent) => void,
+  ){}
+  abstract name: string;
+  status: ResourceHealth = ResourceHealth.Unhealthy;
+  details: string | undefined;
+  abstract checkHealth(): Promise<void> | void;
+}
```

Comment:
We don't use any class inheritance yet and should continue to avoid it unless there's a compelling reason to introduce it.

I think this can just be a type:

```ts
type HealthIndicator = {
  name: string;
  status: ResourceHealth;
  details?: string;
  checkHealth(): Promise<void> | void;
};
```

Then you can write:

```ts
class ExampleHealthIndicator implements HealthIndicator { ... }
```

Instead of:

```ts
class ExampleHealthIndicator extends HealthIndicator { ... }
```

Related: https://www.youtube.com/watch?v=hxGOiiR9ZKg